datasets/cfq/data_generation.py

```diff
@@ -0,0 +1,341 @@
+""" Generating training and test files for iterative decoding on CFQ.
+"""
+
+import csv
+import numpy as np
+import random as rd
+
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets as tfds
+
+
+START_TOKEN = "[START]"
+SEP_TOKEN = "[SEP]"
+IN_OUT_TOKEN = "[SEP2]"
+END_TOKEN = "[END]"
+END_ITER_TOKEN = "[ENDITER]"
+
+
+dict_of_args = {}
+for i in range(10):
+    dict_of_args["M" + str(i)] = ["M" + str(i)]
+
+
+def cfq_decompose_output(line):
+  tokens = line.split(" ")
+  prefix = ""
+  postfix = ""
+  triplets_text = ""
+  state = 0
+  for token in tokens:
+    if state == 0:
+      if token == "{":
+        prefix += token + " "
```

Comment:
Nit: it may be more efficient/typical to append to a list of strings and join them at the end, but I doubt this is causing much of a speed problem so no worries.

(Comment also applies to the `generate_it_dec_examples` function below.)